BeagleBone_black/src/state_machine/main.cpp

```diff
@@ -49,9 +49,19 @@ Main::Main(uint8_t id, Logger& log)
 void Main::run()
 {
   while (1) {
+    data_ = data::Data::getInstance();
+    comms_data_ = data_.getCommunicationsData();
+    nav_data_ = data_.getNavigationData();
+    sm_data_ = data_.getStateMachineData();
+    motor_data_ = data_.getMotorData();
+    batteries_data_ = data_.getBatteriesData();
+    sensors_data_ = data_.getSensorsData();
+
     checkFailure();
     checkCommunications();
```

Comment:
Here you have the same problem as you had with navigation. You cannot access anything other than module statuses before the modules indicate that they have completed initialisation because you would only recieve random garbage.

Reply:
Resolved